src/components/NumberInput/NumberInput.tsx

```diff
@@ -92,7 +92,8 @@ export const NumberInput = ({
         const { key, shiftKey } = event;
         if (SPECIAL_KEYS.includes(key)) {
             const currentValue = getCurrentValueWithoutSuffix();
-            if (currentValue?.length === 0) {
+            const isNegativeSymbol: boolean = currentValue.length === 2 && currentValue.includes('-');
```

Comment:
Can you add a test case to avoid regression?

Reply:
test coverage added